gum/backend-darwin/gumcodesegment-darwin.c

```diff
@@ -221,10 +205,29 @@ gum_code_segment_get_virtual_size (GumCodeSegment * self)
 void
 gum_code_segment_realize (GumCodeSegment * self)
 {
-  gboolean realized_successfully;
+  static gsize realize_supported = 0;
+
+  if (g_once_init_enter (&realize_supported))
+  {
+    gboolean supported = FALSE;
+    gpointer scratch_page;
+    GumCodeSegment * segment;
 
-  realized_successfully = gum_code_segment_try_realize (self);
-  g_assert (realized_successfully);
+    segment = gum_code_segment_new (1, NULL);
+    scratch_page = gum_code_segment_get_address (segment);
+    supported = gum_code_segment_try_realize (segment);
+    if (supported)
+      supported = gum_code_segment_try_map (segment, 0, 1, scratch_page);
+    gum_code_segment_free (segment);
+
+    g_once_init_leave (&realize_supported, supported + 1);
+  }
+
+  if (realize_supported == 2) {
+      gboolean realized_successfully;
```

Comment:
- Brace position needs an adjustment.
- Indentation needs a tweak.
- Blank line after the variable declaration would improve readability a little.